src/main/kotlin/com/github/biomejs/intellijbiome/BiomePackage.kt

```diff
@@ -17,44 +18,47 @@ class BiomePackage(private val project: Project) {
             return NodePackage.findDefaultPackage(project, "@biomejs/biome", interpreter)
         }
 
-    val configPath: String?
-        get() {
-            val settings = BiomeSettings.getInstance(project)
-            val configurationMode = settings.configurationMode
-
-            return when (configurationMode) {
-                ConfigurationMode.DISABLED -> null
-                ConfigurationMode.AUTOMATIC -> null
-                ConfigurationMode.MANUAL -> BiomeSettings.getInstance(project).configPath
-            }
+    fun configPath(file: VirtualFile): String? {
+        val settings = BiomeSettings.getInstance(project)
+        val configurationMode = settings.configurationMode
+        return when (configurationMode) {
+            ConfigurationMode.DISABLED -> null
+            ConfigurationMode.AUTOMATIC -> findPathUpwards(file, configValidExtensions.map { "$configName.$it" })?.path
+            ConfigurationMode.MANUAL -> settings.configPath
         }
+    }
 
     fun versionNumber(): String? {
         val settings = BiomeSettings.getInstance(project)
         val configurationMode = settings.configurationMode
         return when (configurationMode) {
             ConfigurationMode.DISABLED -> null
             ConfigurationMode.AUTOMATIC -> nodePackage?.getVersion(project)?.toString()
-            ConfigurationMode.MANUAL -> getBinaryVersion(binaryPath())
+            ConfigurationMode.MANUAL -> getBinaryVersion(binaryPath(null, true))
         }
     }
 
-    fun binaryPath(): String? {
+    fun binaryPath(configPath: String?, showVersion: Boolean): String? {
         val settings = BiomeSettings.getInstance(project)
         val configurationMode = settings.configurationMode
         return when (configurationMode) {
             ConfigurationMode.DISABLED -> null
-            ConfigurationMode.AUTOMATIC -> nodePackage?.getAbsolutePackagePathToRequire(project)?.let {
-                Paths.get(
-                    it,
-                    "bin/biome"
-                )
-            }?.toString()
-
-            ConfigurationMode.MANUAL -> settings.executablePath
+            // don't try to find the executable path if the configuration file does not exist.
+            // This will prevent start LSP and formatting in case if biome is not used in the project.
+            ConfigurationMode.AUTOMATIC -> if (configPath != null || showVersion) findBiomeExecutable() else null
+            // if configuration mode is manual, return the executable path if it is not empty string.
+            // Otherwise, try to find the executable path.
+            ConfigurationMode.MANUAL -> if (settings.executablePath == "") findBiomeExecutable() else settings.executablePath
         }
     }
 
+    private fun findBiomeExecutable() = nodePackage?.getAbsolutePackagePathToRequire(project)?.let {
+        Paths.get(
+            it,
+            "bin/biome"
```

Comment:
Shouldn't be `.bin/biome`?

Reply:
This will search for `@biomejs/biome/bin/biome`, as nodePackage is nodePackage is already founded `@biomejs/biome` package

Reply:
The Biome binary isn't in that folder.

`@biomejs/biome` loads the correct binary based on the architecture of the machine.

The binaries are distributed in different packages: https://www.npmjs.com/search?q=%40biomejs%2Fcli

Reply:
Yes, but it has biome script inside that routes to proper binary. BTW it's not something that I changed, this code was already in plugin, just moved it to function